.github/workflows/pr.yml

```
@@ -10,13 +10,14 @@ jobs:
 
     steps:
       - name: Checkout the repository
-        uses: actions/checkout@v2.3.4
+        uses: actions/checkout@v4
 
-      - name: Set up node using nvm
-        uses: dcodeIO/setup-node-nvm@v4
+      - uses: actions/setup-node@v4
+        with:
+          node-version: "22"
```

This should be reading the .nvmrc file (also please update .nvmrc version)
```suggestion
      - uses: actions/setup-node@v4
        with:
          node-version-file: ".nvmrc"
```

---

Thanks for your quick review 🙏🏻 

* I have identified other issues I will include in the PR.
* I am not very experienced with the Node ecosystem, and I will need assistance for the Jest part 😅